spirv_msl.cpp

```diff
@@ -3118,7 +3224,7 @@ string CompilerMSL::unpack_expression_type(string expr_str, const SPIRType &type
 	};
 
 	// std140 array cases for vectors.
-	if (physical_type && is_vector(*physical_type) && is_array(*physical_type) && physical_type->vecsize > type.vecsize)
+	if (physical_type && is_vector(*physical_type) && is_array(*physical_type) && physical_type->vecsize > type.vecsize && !expression_ends_with(expr_str, swizzle_lut[type.vecsize - 1]))
```

Comment:
Why is it necessary to string compare here?

Reply:
I think that check was to avoid duplicate swizzle operators like `v.xy.xy`.
The string comparison can handle input expressions in `expr_str` that are unknown at that point. And the maximum string it is compared to is `".xyz"` so that shouldn't be a performance issue.